src/librustc_typeck/diagnostics.rs

```diff
@@ -1130,6 +1244,96 @@ impl Bytes { ... } // error, same as above
 ```
 "##,
 
+E0117: r##"
+You got this error because because you tried to implement a foreign
+trait on a foreign (with maybe a foreign type parameter). Erroneous
```

Comment:
Yes, don't worry I'll. I just didn't want to start fixing typo if I had to come back then to change something else (it will be a very big squash party tonight !).